src/components/flat-table/flat-table.test.tsx

```diff
@@ -0,0 +1,1480 @@
+));
+
+test("should override Pager's top border styling so it connects to the table when passed as `footer`", () => {
+  render(
+    <FlatTable footer={<Pager data-role="pager" onPagination={() => {}} />}>
+      <FlatTableHead>
+        <FlatTableRow>
+          <td>heading one</td>
+        </FlatTableRow>
+      </FlatTableHead>
+      <FlatTableBody>
+        <FlatTableRow>
+          <FlatTableCell>item one</FlatTableCell>
+        </FlatTableRow>
+      </FlatTableBody>
+    </FlatTable>
+  );
+  const pager = screen.getByTestId("pager");
+
+  expect(pager).toHaveStyle({
+    borderTop: "none",
+    borderTopLeftRadius: 0,
+    borderTopRightRadius: 0,
+  });
+});
+
+test("should render with neither the table or Pager with bottom rounded corners when passed as a footer and `hasStickyFooter` is set", () => {
+  render(
+    <FlatTable
+      hasStickyFooter
+      footer={<Pager data-role="pager" onPagination={() => {}} />}
+    >
+      <FlatTableHead>
+        <FlatTableRow>
+          <FlatTableHeader>heading one</FlatTableHeader>
+        </FlatTableRow>
+      </FlatTableHead>
+      <FlatTableBody>
+        <FlatTableRow>
+          <FlatTableCell>child one</FlatTableCell>
+        </FlatTableRow>
+      </FlatTableBody>
+    </FlatTable>
+  );
+  const tableWrapper = screen.getByRole("region");
+  const pager = screen.getByTestId("pager");
+
+  expect(tableWrapper).toHaveStyle({
+    borderBottomLeftRadius: undefined,
+    borderBottomRightRadius: undefined,
+  });
+  expect(pager).toHaveStyle({
+    borderBottomLeftRadius: 0,
+    borderBottomRightRadius: 0,
+  });
+});
+
+describe("when rows are interactive", () => {
+  it("should not move focus to first row with `onClick` when down arrow pressed and table wrapper focused when `focusRedesignOptOut` is not set", async () => {
+    const user = userEvent.setup();
+    render(
+      <FlatTable>
+        <FlatTableBody>
+          <FlatTableRow onClick={() => {}}>
+            <FlatTableCell>one</FlatTableCell>
+            <FlatTableCell>two</FlatTableCell>
+          </FlatTableRow>
+          <FlatTableRow onClick={() => {}}>
+            <FlatTableCell>three</FlatTableCell>
+            <FlatTableCell>four</FlatTableCell>
+          </FlatTableRow>
+        </FlatTableBody>
+      </FlatTable>
+    );
+    const tableWrapper = screen.getByRole("region");
+    const focusableTableContainer = screen.getByTestId("flat-table-container");
+    focusableTableContainer.focus();
+
+    expect(focusableTableContainer).toHaveFocus();
+    expect(tableWrapper).toHaveStyleRule("outline", "transparent 3px solid");
+    expect(tableWrapper).toHaveStyleRule(
+      "box-shadow",
+      "0px 0px 0px var(--borderWidth300) var(--colorsSemanticFocus500),0px 0px 0px var(--borderWidth600) var(--colorsUtilityYin090)"
+    );
+
+    await user.keyboard("ArrowDown");
+
+    expect(focusableTableContainer).toHaveFocus();
+    expect(screen.getByRole("row", { name: "one two" })).not.toHaveFocus();
+  });
```

Comment:
I'm happy to do it but I'm not convinced there's much difference between the two implementations wrt to testing them independently as once play functions are implemented the styling part will be deleted. I will accept that it will be easier to remove if they're in isolation